docs/developing_aci_module_guidelines.md

```diff
@@ -0,0 +1,700 @@
+
+Once the AnsibleModule object is initialized as module, retrieve values from the module.params dictionary. Typically, only parameters related to the ACI object and its child configuration need to be extracted. Any necessary type checks or validations (e.g., capitalize the input or add special characters) should also be performed here.
+
+```python
+object_id = module.params.get('object_id')
+object_prop1 = module.params.get('object_prop1')
+object_prop2 = module.params.get('object_prop2')
+object_prop3 = module.params.get('object_prop3')
+if object_prop3 is not None and object_prop3 not in range(x, y):
+    module.fail_json(msg='Valid object_prop3 values are between x and (y-1)')
+child_object_id = module.params.get('child_object_id')
+child_object_prop = module.params.get('child_object_prop')
+state = module.params.get("state")
+```
+
+> [!NOTE]
+> * In some cases, the APIC requires special characters (e.g., [, ], or -) in names, or uses internal metadata (e.g., "port_binding" attribute in aci_epg_to_domain module). Modules should handle formatting or concatenation of parameters internally to keep user input simple and intuitive.
+> * Most type conversions and validations at this stage are minimal and are intended to ensure that properly formatted data is passed into subsequent API calls or logic.
+> * Certain additional validations are only added when API response for these checks are not as expected.
+
+### Using the ACIModule object
+
+The ACIModule class manages most of the logic used by ACI modules. It extends the functionality of the AnsibleModule object, so the module instance must be passed during instantiation:
+
+```python
+aci = ACIModule(module)
+```
+
+The ACIModule includes 7 main methods commonly used across modules:
+
+* construct_url
+* get_existing
+* payload
+* get_diff
+* post_config
+* delete_config
+* exit_json
+
+The first 2 methods are used regardless of what value is passed to the `state` parameter.
+
+#### Constructing URLs
+
+The `construct_url()` method dynamically builds the REST API URL and query parameters for retrieving or configuring ACI objects. It supports multiple levels of the object hierarchy by accepting up to six optional subclass dictionaries and one list of child classes.
+
+This method uses the root class (e.g., fvTenant) and optional subclass dictionaries to construct the full distinguished name (DN) of the object. It also applies filters and response modifiers.
+
+* When the `state` is not `query`:The URL includes the full DN (base URL (to access the APIC) with the distinguished name of the object (to access the object)) of the object, and the response is typically limited to configuration data (config-only).
+* When `state` is `query`, the URL and filter string used depend on which parameters are passed to the object. This method handles the complexity so that it is easier to add new modules and ensures that all modules are consistent in the type of data returned.
+
+    > [!IMPORTANT]
```

Comment:
All the snippets seem to be fixed except this one ->
<img width="966" height="135" alt="Screenshot 2025-07-29 at 5 03 31 PM" src="https://github.com/user-attachments/assets/f1a8f364-a376-43f7-abc7-aa318c4eedd7" />